protocol/communities/manager.go

```diff
@@ -876,7 +871,11 @@ func (m *Manager) HandleCommunityRequestToJoin(signer *ecdsa.PublicKey, request
 		return nil, err
 	}
 
-	if community.AcceptRequestToJoinAutomatically() {
+	// If user is already a member, then accept request automatically
+	// It may happen when member removes itself from community and then tries to rejoin
+	// More specifically, CommunityRequestToLeave may be delivered later than CommunityRequestToJoin, or not delivered at all
+	acceptAutomatically := community.AcceptRequestToJoinAutomatically() || community.HasMember(signer)
+	if acceptAutomatically {
```

Comment:
Just to raise awareness, this will trigger a `AcceptRequestToJoinCommunity` signal, which might not be wanted, given that with this condition, the requester is already a member.

Reply:
Thanks, I am aware of that. As stated in the comment above, this logic is expected. If the user is a member and asks to join the community, the response should be positive, because we can't know if `RequestToLeave` wasn't sent before that.

See "Considerations" section from the issue description: https://github.com/status-im/status-go/issues/2812

Reply:
Correct. Just wanted to make sure we double check in clients (Desktop) that `AcceptRequestToJoin` won't cause funny results.

Feel free to ignore this :)

Reply:
What you say is hard to reproduce, because once you join the community the UI doesn't give you the ability to join again, unless you leave. The basic flow (join, leave, join) was tested both on Desktop and Mobile. Hope there is no any edge case I've missed, but if there is, it would be fixed as a separate issue I guess.